src/main/java/org/broadinstitute/hellbender/utils/read/ArtificialBAMBuilder.java

```diff
@@ -0,0 +1,222 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.reference.IndexedFastaSequenceFile;
+import org.broadinstitute.hellbender.utils.GenomeLocParser;
+import org.broadinstitute.hellbender.utils.NGSPlatform;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Easy to use creator of artificial BAM files for testing
+ *
+ * Allows us to make a stream of reads or an index BAM file with read having the following properties
+ *
+ * - coming from n samples
+ * - of fixed read length and aligned to the genome with M operator
+ * - having N reads per alignment start
+ * - skipping N bases between each alignment start
+ * - starting at a given alignment start
+ */
+public final class ArtificialBAMBuilder {
+    public static final int BAM_SHARD_SIZE = 16384;
+
+    private final IndexedFastaSequenceFile reference;
+    private final GenomeLocParser parser;
+
+    final int nReadsPerLocus;
+    final int nLoci;
+
+    int skipNLoci = 0;
+    int alignmentStart = 1;
+    int readLength = 10;
+    private final ArrayList<String> samples = new ArrayList<>();
+    private List<GATKRead> createdReads = null;
+
+    private List<GATKRead> additionalReads = new LinkedList<>();
+    private SAMFileHeader header;
+
+    final SAMFileWriterFactory factory = new SAMFileWriterFactory().setCreateIndex(true);
```

Comment:
Use `ReadUtils.createCommonSAMWriter()` (we are trying to centralize creation of these factories to ensure support for cram, among other reasons).